tests/mocks/Matchers.cpp

```diff
@@ -163,14 +163,18 @@ bool PayloadMatcher::MatchAndExplain(
   }
 
   bool matched = true;
+  unsigned int block_offset = 0;
+  std::ios::fmtflags ostream_flags;
   if (listener->IsInterested()) {
-    unsigned int block_offset = 0;
-    std::ios::fmtflags ostream_flags(listener->stream()->flags());
-    for (unsigned int i = 0; i < m_expected_size; i++) {
-      uint8_t actual = reinterpret_cast<const uint8_t*>(
-          iovec->base)[block_offset];
-      uint8_t expected = m_expected_data[i];
+    ostream_flags = listener->stream()->flags();
+  }
+
+  for (unsigned int i = 0; i < m_expected_size; i++) {
+    uint8_t actual = reinterpret_cast<const uint8_t*>(
+        iovec->base)[block_offset];
+    uint8_t expected = m_expected_data[i];
 
+    if (listener->IsInterested()) {
       *listener
          << "\n" << std::dec << i << ": 0x" << std::hex
```

Comment:
Okay. I figured given we've got it. And you've basically just rewritten the byte dump function again...